### ISSUE TITLE
EPIC: Clean up Nagios so that we can use #infrastructure-bot

### ISSUE THREAD

**Issue by sxa** [adoptium/infrastructure#2445]
Currently working on the following tasks related to this issue:

Issue #2609 - PR #2638 Awaiting Review - Completed (29/09/2022 )
1) Create ansible playbook to install Nagios Server Core on Ubuntu 22.04 via Ansible

Issue [#2650](https://github.com/adoptium/infrastructure/issues/2650) - Completed ( 29/09/2022 )  
2) Tidy Up Existing Nagios Processeses, To Provide Base For Future Automation

Issue [#2619](https://github.com/adoptium/infrastructure/issues/2619) - Completed ( 27/06/2022 )  
3) Upgrade O/S on Nagios Production Server To Ubuntu 22.04

Issue #2758 - Completed ( 29/09/2022 )  
4) Align Nagios & Ansible Inventory Files, Fix Hosts In Nagios But Not Ansible

Issue #2759 - Completed ( 22/12/2022 )  
5) Align Nagios & Ansible Inventory Files, Fix Hosts In Ansible But Not Nagios

PR https://github.com/adoptium/infrastructure/pull/2948
6) Automate Creation Of Nagios Configuration Elements From Ansible Inventory

Issue [#2629](https://github.com/adoptium/infrastructure/issues/2629)  - Draft PR Ready(31/01/2023)
6.1) Automate parsing of ansible inventory into Nagios inventory
Issue [#2864](https://github.com/adoptium/infrastructure/issues/2864) 
6.2) Automate parsing of ansible inventory to create hostgroups.cfg - Completed (10/01/2023)
Issue [#2876](https://github.com/adoptium/infrastructure/issues/2876) 
6.3) Automate parsing of ansible inventory to create servicegroups.cfg
Issue [#2865](https://github.com/adoptium/infrastructure/issues/2865) - Completed (27/01/2023)
6.4) Automate population of machine configurations using service/types and j2 template mappings..
6.4.1 ) Windows - Issue #2899 
PR - https://github.com/adoptium/infrastructure/pull/2948

Issue [#2802](https://github.com/adoptium/infrastructure/issues/2802) - Completed (21/11/2022)
7.1) Investigate Automatic Adding Of Windows Hosts To Nagios & Document
-- Need to create new template, windows version of remote tunnel etc.

Issue [#2825](https://github.com/adoptium/infrastructure/issues/2825) - Completed ( 02/12/2022 )  
7.2) Update Ansible Windows Playbook To Install NSClient Required By Nagios
PR: https://github.com/adoptium/infrastructure/pull/2827

Issue [#1718](https://github.com/adoptium/infrastructure/issues/1718) - Completed ( 02/12/2022 )  
7.3) Add Windows Machines To Nagios

8) Add additional monitoring for key infrastructure machines & dockerhosts ( these may need specialist individual setups and configurations creating to monitor system unique services etc )

Issue #2587 -  Completed 16/12/2022
8.1) Configure dockerhosts to be monitored in Nagios
Issue #2903 - Completed 02/2/2023
8.2) Configure dynamic docker-xxxx hosts to be monitored in Nagios

8.3)  Expand Dockerhost Monitoring For Container/Workspace Usage - Requires New Nagios Plugin To Be Written
Issue #2929 - Completed 01/2023
8.4)  Expand Dockerhost Monitoring To Check For Non Running Containers - Requires New Nagios Plugin To Be Written

8.5) Configure Unique Systems
List Of Unique Systems:
     **trss.adoptopenjdk.net - Completed**
     **nagios.adoptopenjdk.net - Completed** 
     **ci.adoptopenjdk.net- Completed** 

8.5.1) Configure TRSS - #2904 - Completed 02/12/2022 
8.5.2) Configure Jenkins - #2905 - Completed 02/02/2023

PR https://github.com/adoptium/infrastructure/pull/2948 delivers these elements
9) Automate Installation & Configuration Of Nagios Nodes ( Requires 2 & 4 )  
9.2) Update server related global config files ( nagios.cfg )  
9.3) Create Commands.CFG file from source. ( commands.cfg )

10) Additional Tasks Carried Out:-
10.1) Document process to update host group : https://github.com/adoptium/infrastructure/issues/2765
10.2) Document process to add additional host check : https://github.com/adoptium/infrastructure/issues/2767

11) Review and standardise host checks
11.1) Issue #2199  - Standardise & Rework The Timesync Role - Completed ( 02/2023 )
11.2) Amend check_ntp_timesync script to work for both ntpd and chronyd services - Completed ( 02/2023 )

---

We have a slack channel for Nagios notifications (#infrastructure-bot) but it's a bit noisy, mostly due to some things like NTP not being in a suitable state. At the moment the channel is receiving hundreds of notifications a day which prevents it being useful. We need to clean up this output (fix the problems, or choose not to publish the notifications) so that we can take remedial action in a timely manner to reduce the impact to the build and test processes. This epic will have multiple individual issues underneath it to cover things that can be tackled individually.

Follow on to #1229

**Comment by Willsparker** [comment]
I'd like to propose we change this EPIC from cleaning up Nagios specifically so we can use #infrastructure-bot , to cleaning up Nagios, in it's entirety - i.e. Upgrade the 

Currently; Nagios 4.4.6 (the latest Nagios) is running on an older version of Ubuntu on Hetzner. With the recent look at upgrading our Jenkins instance, it may be a good idea to do the same with the Nagios machine. 

In addition, Nagios was initially setup by Brad, and using a shell script that was called by Ansible, however no documentation was made about why certain checks have been put in. I attempted to clean it up last year and made a lot of issues for future enhancements, but it may be best to just completely re-set it up with; documentation for why we do each check, a proper Ansiblilised method of adding each host to Nagios (which, with AWX, would make keeping the hosted machine on Nagios much easier to keep updated), and very well documenting any manual setup, if required, for specific hosts (i.e. The `check_SSL_cert` check would likely need to be put in manually). 

Over the next little while, I'll do some local testing on my machine to ansibilise the Nagios Host setup process (i.e. replacing the script Brad made), to make a start.

**Comment by Willsparker** [comment]
ref: #1716 #1718 #2039 #1876 #1670

(there may be more, to be honest)

**Comment by Willsparker** [comment]
Ref: https://github.com/Willsparker/AnsibleBoilerPlates/tree/main/Nagios
I've created a playbook that can install Nagios-Core on at least Ubuntu 20.04 (I imagine it'd work with 21.04 too). It isn't perfect, but it'll help if we do a big new install of Nagios (and also makes it easy to create a test bed to help with #1670 ).

**Comment by steelhead31** [comment]
Currently working on the following tasks related to this issue:

1) Create ansible playbook to install Nagios Server Core on Ubuntu 22.04 via Ansible
2) Create ansible playbook to install Nagios on Ubuntu 22.04 Host To Be Monitored
3) Create ansible playbook to add installed monitor nodes on Nagios server

Other tasks to be considered :

1) Document standard tasks provided on base install
2) Identify additional monitoring tasks, and automate adding them via ansible.

**Comment by steelhead31** [comment]
Add checks for Vagrant host, specifically around number of network connections and zombie boxes.

**Comment by sxa** [comment]
From discussion today with @sej-jackson and @aixtools - feel free to add anything I've missed in subsequent comments as my notes didn't get saved so I'm now writing this from memory:

- Add monitoring of `/tmp` and the file system used for the jenkins workspace (Was noted that if workspace gets full, jenkins will mark the machine offline which will be picked up as a warning space regardless)
- Potential to replace some of the items from the [Process Check](https://ci.adoptopenjdk.net/view/Tooling/job/SXA-processCheck/) job - in particular causing a warning alert if there is an `X -vfb` process which is over 24 hours old which is running as the jenkins user. As a second stage, potentially anything other than the jenkins agent process should be treated as suspicious and generate a warning if there are no jobs currently in the process of being executed
- Possibly use to check system logs or processes on the machine for any intrusion attempts

Other systems that could be monitored as part of the infrastructure servers or elsewhere (likely each need custom checks)
- TRSS
- AWX
- Jenkins itself
- Vagrant server (e.g. when it runs low on available network interface numbers to use!)

We also discussed whether it could be used to alert on the things currently monitored by https://status.adoptium.net/

And we also suggested that it might be worth having a special group for systems which are being used as hosts for our 'dockerstatic' containers used for tests to monitor their health, whether the container is up etc. as these are likely to form an increasing proportion of the machines which we use.
